helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -384,8 +384,8 @@ public synchronized void refresh(HelixDataAccessor accessor) {
     _instanceMessagesCache.updateRelayMessages(_liveInstanceCache.getPropertyMap(),
         _currentStateCache.getParticipantStatesMap());
 
-    updateIdealRuleMap();
-    updateDisabledInstances();
+    updateIdealRuleMap(getClusterConfig());
```

Comment:
why here we use getClusterConfig() while later in `setClusterConfig` we use _clusterConfig?

In fact, 

```
  public ClusterConfig getClusterConfig() {
    return _clusterConfig;
  }
```

any specific reason?

Reply:
This seems to be a non-essential change?

Reply:
Because setClusterConfig() is where you update _clusterConfig. And here, it is using it as an input.
It is a good practice that we don't refer to a private field directly in a private method. Note that the main reason no one calls this method to update for now is that we hide it too deep, so no one remembers. If the parameters are explicitly passed, meaning input and output are clearly divided, then we avoid potential bugs.
So, IMO this is tightly related to this change.